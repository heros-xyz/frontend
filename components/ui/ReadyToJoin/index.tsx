import { Box, Text, HStack, Divider, VStack } from "@chakra-ui/react";
import Image from "next/image";

interface BannerProps {
  title: string;
  banner: string;
}
interface ReadyToJoinProp {
  content: {
    title: string;
    description: string;
    banner: {
      fanBanner: string;
      fanTitle: string;
      athleteBanner: string;
      athleteTitle: string;
    };
  };
}

function Banner(props: BannerProps) {
  return (
    <Box>
      <Image src={props?.banner} alt="banner" width={335} height={105} />
      <HStack
        bg={"primary"}
        borderBottomRadius={"10px"}
        height={12}
        align="center"
        justify={"center"}
      >
        <Text
          color={"secondary"}
          fontSize="md"
          fontWeight={700}
          lineHeight="1.4rem"
          fontFamily={"heading"}
        >
          {props?.title.toUpperCase()}
        </Text>
      </HStack>
    </Box>
  );
}
const ReadyToJoin: React.FC<ReadyToJoinProp> = (props) => {
  const { content } = props;
  return (
    <Box
      bg="accent.4"
      maxWidth={375}
      minHeight={528}
      borderTopRightRadius={"18px"}
      borderBottomRadius={"18px"}
    >
      <VStack mx={5} mt={7.5} mb={10}>
        <Text
          fontFamily={"heading"}
          alignSelf={"flex-start"}
          fontSize={"2xl"}
          as="span"
          color={"accent.2"}
          lineHeight={"2.1rem"}
        >
          {content?.title}
        </Text>
        <Text
          fontFamily={"heading"}
          color="primary"
          fontSize={"sm"}
          as="p"
          lineHeight={"19.6px"}
          pt={0.5}
          pb={1.25}
        >
          {content?.description}
        </Text>
        <Banner
          title={content?.banner?.fanTitle}
          banner={content?.banner?.fanBanner}
        />
        <HStack justifyContent={"space-between"} width="100%" py={5}>
          <Divider maxW={"8.438rem"} />
          <Text fontSize={"xl"} fontWeight={800} fontFamily={"heading"}>
            Or
          </Text>
          <Divider maxW={"8.438rem"} />
        </HStack>
        <Banner
          title={content?.banner?.athleteTitle}
          banner={content?.banner?.athleteBanner}
        />
      </VStack>
    </Box>
  );
};

export default ReadyToJoin;
