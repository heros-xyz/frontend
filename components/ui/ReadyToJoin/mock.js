export const mock = {
  content: {
    title: "Are you ready to join?",
    description:
      "Donâ€™t miss this amazing opportunity to shorten the gap between athletes and fans.",
    banner: {
      fanTitle: "become a fan",
      fanBanner: "/images/athlete-banner.png",
      athleteBanner: "/images/fan-banner.png",
      athleteTitle: "become an athlete"
    }
  }
};