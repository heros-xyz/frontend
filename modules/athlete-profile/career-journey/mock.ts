export const CAREER_JOURNEY_DATA = [
  {
    title: "vvvv",
    description: "vvvv",
    from: "2022-11-29",
    to: null,
    isArchive: true,
    isCurrent: true,
    icon: "cricket",
  },
  {
    title: "Ggggh",
    description: "",
    from: "2012-01-24",
    to: "2022-02-20",
    isArchive: true,
    isCurrent: false,
    icon: "flag",
  },
  {
    title: "Hhh",
    description: "Uu",
    from: "1991-03-30",
    to: null,
    isArchive: true,
    isCurrent: false,
    icon: "golf",
  },
  {
    title: "Hh",
    description: "Hhhh",
    from: "2021-03-29",
    to: "2021-03-30",
    isArchive: true,
    isCurrent: false,
    icon: "soccer",
  },
  {
    title: "Ghjkkj ghh",
    description: "Ghi kkk",
    from: "1997-10-29",
    to: "2021-10-29",
    isArchive: true,
    isCurrent: false,
    icon: "soccer",
  },
  {
    title: "Phong",
    description: "",
    from: "1980-02-29",
    to: null,
    isArchive: true,
    isCurrent: false,
    icon: "golf",
  },
  {
    title: "My nam is Phong",
    description:
      "Hhh ahhâj dkf nffn fkfk hâhh dpgong jfjffn bxbxxb abb có gò nưav đâu mà ",
    from: "2022-10-30",
    to: "2023-10-30",
    isArchive: true,
    isCurrent: false,
    icon: "medal",
  },
  {
    title: "Hhh",
    description: "Hhh",
    from: "2021-10-30",
    to: null,
    isArchive: true,
    isCurrent: false,
    icon: "soccer",
  },
];
