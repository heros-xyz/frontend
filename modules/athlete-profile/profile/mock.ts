export const basicInfo = {
  id: "c2b7f17a-ee39-4b69-a35f-4ec8cf6c7dcb",
  createdAt: "2023-01-17T05:46:06.812Z",
  updatedAt: "2023-02-03T11:26:19.748Z",
  deletedAt: null,
  nickName: "dds",
  story:
    "abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL NO PQRS TUV WXYZ !\"§ $%& /() =?* '<> #|; ²³~ @`´ ©«» ¤¼× {} abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL MNO PQRS TUV WXYZ !\"§ $%& /() =?* '<> #|; ²³~ @`´ ©«» ¤¼× {} abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL MNO PQRS TUV WXYZ !\"§ $%& /() =?* '<> #|; ²³~ @`´ ©«» ¤¼× {} abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL MNO PQRS TUV WXYZ !\"§ $%& /() =?* '<> #|; ²³~ @`´ ©«» ¤¼× {} abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL MNO PQR",
  nationality: {
    id: "58cd8f52-76b0-46dd-9a29-8cef2dbcfc74",
    createdAt: "2023-01-16T10:22:40.062Z",
    updatedAt: "2023-01-16T10:22:40.062Z",
    deletedAt: null,
    name: "Albanian",
    code: "8",
    phoneCode: "355",
  },
  dateOfBirth: "2022-11-29T00:00:00.000Z",
  gender: 0,
};

export const sportProfile = {
  data: {
    id: "dc3fb3e3-b659-427f-875b-4fe5549c71fb",
    createdAt: "2023-01-17T08:49:40.080Z",
    updatedAt: "2023-01-17T08:49:40.080Z",
    sportProfilesItems: [],
    goal: "fff",
    currentTeam: "dsadasa",
  },
  success: true,
};

export const basicTest = {
  id: "c2b7f17a-ee39-4b69-a35f-4ec8cf6c7dcb",
  createdAt: "2023-01-17T05:46:06.812Z",
  updatedAt: "2023-02-03T11:26:19.748Z",
  deletedAt: null,
  nickName: "dds",
  story:
    "abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL NO PQRS TUV WXYZ !\"§ $%& /() =?* '<> #|; ²³~ @`´ ©«» ¤¼× {} abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL MNO PQRS TUV WXYZ !\"§ $%& /() =?* '<> #|; ²³~ @`´ ©«» ¤¼× {} abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL MNO PQRS TUV WXYZ !\"§ $%& /() =?* '<> #|; ²³~ @`´ ©«» ¤¼× {} abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL MNO PQRS TUV WXYZ !\"§ $%& /() =?* '<> #|; ²³~ @`´ ©«» ¤¼× {} abc def ghi jkl mno pqrs tuv wxyz ABC DEF GHI JKL MNO PQR",
  nationality: {
    id: "58cd8f52-76b0-46dd-9a29-8cef2dbcfc74",
    createdAt: "2023-01-16T10:22:40.062Z",
    updatedAt: "2023-01-16T10:22:40.062Z",
    deletedAt: null,
    name: "Albanian",
    code: "8",
    phoneCode: "355",
  },
  dateOfBirth: "2022-11-29T00:00:00.000Z",
  gender: 0,
  firstName: "a",
  lastName: "b",
};
