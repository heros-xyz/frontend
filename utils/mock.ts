export const notiList = [
  {
    id: "1",
    type: "A_NEW_INTERACTION",
    message: "notification.athlete-new-interaction",
    source: {
      id: "48",
      avatar: "https://cdn-icons-png.flaticon.com/512/3135/3135715.png",
      fullName: "Sam Smith",
    },
    interaction: {
      id: "70",
      content: "some content should be trimmed if too long",
    },
    comment: null,
    readAt: null,
    createdAt: "2023-02-22T08:43:57.555Z",
  },
  {
    id: "3",
    type: "A_NEW_INTERACTION",
    message: "notification.athlete-new-interaction",
    source: {
      id: "4",
      avatar: "https://cdn-icons-png.flaticon.com/512/3135/3135715.png",
      fullName: "Sam Smith",
    },
    interaction: {
      id: "35",
      content: "some content should be trimmed if too long",
    },
    comment: null,
    readAt: null,
    createdAt: "2023-02-22T08:43:57.555Z",
  },
];

export const fanNotiList = [
  {
    id: "1",
    type: "A_NEW_INTERACTION",
    message: "notification.athlete-new-interaction",
    source: {
      id: "48",
      avatar: "https://cdn-icons-png.flaticon.com/512/3135/3135715.png",
      fullName: "Sam Smith",
    },
    interaction: {
      id: "70",
      content: "some content should be trimmed if too long",
    },
    comment: null,
    readAt: null,
    createdAt: "2023-02-22T08:43:57.555Z",
  },
  {
    id: "2",
    type: "A_NEW_INTERACTION",
    message: "notification.athlete-new-interaction",
    source: {
      id: "4",
      avatar: "https://cdn-icons-png.flaticon.com/512/3135/3135715.png",
      fullName: "Sam Smith",
    },
    interaction: {
      id: "35",
      content: "some content should be trimmed if too long",
    },
    comment: null,
    readAt: null,
    createdAt: "2023-02-22T08:43:57.555Z",
  },
];

export const sports = [
  {
    value: "Football",
    label: "Football",
  },
  {
    value: "Basketball",
    label: "Basketball",
  },
  {
    value: "Badminton",
    label: "Badminton",
  },
  {
    value: "Cricket",
    label: "Cricket",
  },
  {
    value: "Swimming",
    label: "Swimming",
  },
  {
    value: "Tennis",
    label: "Tennis",
  },
  {
    value: "Baseball",
    label: "Baseball",
  },
  {
    value: "Running",
    label: "Running",
  },
  {
    value: "Flexing",
    label: "Flexing",
  },
];

export const timelineItems = [
  {
    title: "Joined Jaworznickie Stowarzyszenie Pi≈Çkarskie Clubs",
    description:
      "I became the best new player on team and played with team about 10 champion league. We won 8 times .",
    from: "10/01/2022",
    to: "10/24/2022",
    isArchive: true,
    isCurrent: true,
    icon: "flag",
  },
  {
    title: "Join England Club",
    description: "I became the best new player on team",
    from: "10/24/2022",
    isArchive: false,
    isCurrent: false,
  },
  {
    title: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
    description:
      "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
    from: "10/01/2022",
    to: "10/24/2022",
    isArchive: true,
    isCurrent: true,
    icon: "flag",
  },
];
