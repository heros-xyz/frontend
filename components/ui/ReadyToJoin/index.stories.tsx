import { HStack } from "@chakra-ui/react";
import type { Meta, StoryObj } from "@storybook/react";
import ReadyToJoin from ".";
import { mock } from "./mock";

const meta: Meta<typeof ReadyToJoin> = {
  title: "Components/ReadyToJoin",
  component: ReadyToJoin,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
};

export default meta;

type Story = StoryObj<typeof ReadyToJoin>;

export const ReadyToJoinComponent: Story = {
  args: mock,
  render: (args) => (
    <HStack bg="primary" align={"center"} justify="center" padding={10}>
      <ReadyToJoin {...args} />
    </HStack>
  ),
};
