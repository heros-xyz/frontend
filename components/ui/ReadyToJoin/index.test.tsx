import { render } from "@testing-library/react";
import ReadyToJoin from ".";
import { mock } from "./mock";

it("renders ReadyToJoin unchanged", () => {
  const { container } = render(<ReadyToJoin {...mock} />);
  expect(container).toMatchSnapshot();
});
