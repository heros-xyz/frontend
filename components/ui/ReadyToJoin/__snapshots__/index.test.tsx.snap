// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ReadyToJoin unchanged 1`] = `
<div>
  <div
    class="css-laphny"
  >
    <div
      class="chakra-stack css-7ejnlj"
    >
      <span
        class="chakra-text css-1we3h2x"
      >
        Are you ready to join?
      </span>
      <p
        class="chakra-text css-1c9x3vt"
      >
        Donâ€™t miss this amazing opportunity to shorten the gap between athletes and fans.
      </p>
      <div
        class="css-0"
      >
        <img
          alt="banner"
          data-nimg="1"
          decoding="async"
          height="105"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fathlete-banner.png&w=750&q=75"
          srcset="/_next/image?url=%2Fimages%2Fathlete-banner.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fathlete-banner.png&w=750&q=75 2x"
          style="color: transparent;"
          width="335"
        />
        <div
          class="chakra-stack css-jel4q2"
        >
          <p
            class="chakra-text css-169w3k9"
          >
            BECOME A FAN
          </p>
        </div>
      </div>
      <div
        class="chakra-stack css-1amziim"
      >
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-z4i0b0"
        />
        <p
          class="chakra-text css-aetsi8"
        >
          Or
        </p>
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-z4i0b0"
        />
      </div>
      <div
        class="css-0"
      >
        <img
          alt="banner"
          data-nimg="1"
          decoding="async"
          height="105"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Ffan-banner.png&w=750&q=75"
          srcset="/_next/image?url=%2Fimages%2Ffan-banner.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Ffan-banner.png&w=750&q=75 2x"
          style="color: transparent;"
          width="335"
        />
        <div
          class="chakra-stack css-jel4q2"
        >
          <p
            class="chakra-text css-169w3k9"
          >
            BECOME AN ATHLETE
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
